### ISSUE TITLE
Please add levels.

### ISSUE THREAD

**Issue by Husnain1** [huytd/agar.io-clone#335]

**Comment by abalabahaha** [comment]
What do you mean by levels?

**Comment by Husnain1** [comment]
I don't know programming but i can give you best ideas like add XP, Coins, items, Facebook login etc...
![untitled](https://cloud.githubusercontent.com/assets/15088002/10455794/b870ea30-71bd-11e5-94d2-d56d6b5b008e.png)

**Comment by Logan7432** [comment]
yeah! to be honest this would be cool! i see some server have this and it would be great to have this!

**Comment by Husnain1** [comment]
Are you a professional web developer?

**Comment by Logan7432** [comment]
Not exactly but I'm learning how to do it.

**Comment by Husnain1** [comment]
Can you add markets, Items, coins etc... in this game? and can you develop this game in flash and playerio (gamesnet.yahoo.com)

**Comment by 3642066** [comment]
When Agario implemented Facebook they walked into a great step on getting new users, and when they starting using XP the users come back, same score of before.. So, they stimulate login giving an starting mass for the users, and unfortunally turn to another way when started that monetizing everething all over the game. They could use more frictionless actions for sharing actions, activities, scores and exchange social actions into game store products, for keep development going ahead instead of the most commercial solluction possible. Anyway, they sill can socialize the game, and if that game platform help on that, it will be an awesome update . 
For this project, i think Facebook Login is nice for example, (as option) for giving names to the cells, or maybe drawing the profile picture as an skin? Yes its possible .
Its also good for in case of testing, no players are online, right ? 
Using Facebook we can send invites for other users, and that could return as many things . 
Free mass, speed, but that need to be implemented into final game development stage .

**Comment by 3642066** [comment]
Some weeks ago i saw some interesting projects of game store, that i think it could be implemented here. Do you have any other ideas of Items that could be offered ? 
As original game have "Starting mass", "Double XP" and "Skins", what else could we offer ?
One more question : 
Why the fuck you want that a HTML5 Canvas game to be redeveloped using Flash ?!
Are you sure you really really wanna that ?

**Comment by Husnain1** [comment]
"3642066" Are you a developer?

**Comment by ThatGamerBlue** [comment]
No, we're not moneyclip.

**Comment by CoderJoeW** [comment]
You can assign this issue to me and I will start working on implementing this.

**Comment by CoderJoeW** [comment]
My thoughts on this are using an external mariadb instance(technically it could be on the same host)

I think it would be easier to manage user logins/levels/etc in a mariadb instance

Maybe we can use the config to implement support for multiple different databases(mysql, mariadb,postgresql, redis)

What I dont want to do is break that deployment to heroku which I am actually not that familiar with.

I am drafting up a branch with some basic support at the moment but if anyone has some insight into if mariadb can be deployed with heroku or if it has to be external or maybe we should make this an optional feature

Opinions welcome
